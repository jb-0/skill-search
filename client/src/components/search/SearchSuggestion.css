.suggested-term-container {
  width: 250px;
  background-color: var(--secondary-color);
  margin: auto;
  padding: 10px 0;
  border: solid;
  border-color: var(--primary-color);
  border-width: thin;
}

.suggested-term-container-mobile {
  width: 50%;
}

.suggested-term-container p {
  margin: 0;
}

.suggested-term-container:hover {
  color: var(--tertiary-color);
  background-color: var(--primary-color);
}

.suggested-term-container:hover p {
  color: var(--tertiary-color);
}