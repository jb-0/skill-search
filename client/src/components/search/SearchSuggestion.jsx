import React, { useContext } from 'react';
import './SearchSuggestion.css';
import { v4 as uuidv4 } from 'uuid';
import { ViewContext } from '../../context/ViewContext';
import { SearchContext } from '../../context/SearchContext';

function SearchSuggestion() {
  const size = useContext(ViewContext);
  const [search, setSearch]  = useContext(SearchContext);

  // Add items to the search terms array
  function addSearchTerm(event) {
    const term = event.target.id;

    setSearch((previousValues) => {
      // Only add the item if it is not already included
      if (!(previousValues.searchTerms.includes(term))) {
        return {
          ...previousValues,
          searchTerms: [...previousValues.searchTerms, term],
        };
      } else {
        return previousValues;
      }
    });
  }

  return (
    <div>
      {search.suggestedTerms.map((suggestedTerm) => {
        return (
          <div
            onClick={addSearchTerm}
            id={suggestedTerm}
            className={
              size.device === 'Desktop'
                ? 'suggested-term-container'
                : 'suggested-term-container-mobile suggested-term-container'
            }
            key={uuidv4()}
            data-testid="suggested-term-container"
          >
            <p id={suggestedTerm}>{suggestedTerm}</p>
          </div>
        );
      })}
    </div>
  );
}

export default SearchSuggestion;
