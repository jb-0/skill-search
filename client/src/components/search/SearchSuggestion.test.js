import '@testing-library/jest-dom/extend-expect';
import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";

import SearchSuggestion from "./SearchSuggestion";

describe('SearchTerms component', () => {
  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('suggested terms passed in via array are rendered successfully', () => {
    act(() => {
      render(<SearchSuggestion suggestedTerms={['node', 'sql']} />, container);
    });

    expect(screen.getByText('node')).toBeInTheDocument();
    expect(screen.getByText('sql')).toBeInTheDocument();
  });

  it('when an empty array is passed, no suggested terms are rendered', () => {
    act(() => {
      render(<SearchSuggestion suggestedTerms={[]} />, container);
    });

    expect(screen.queryAllByTestId('suggested-term-container')).toHaveLength(0);
  });

  it('renders correctly when passed an array containing suggested terms', () => {
    const tree = renderer
      .create(<SearchSuggestion suggestedTerms={['node', 'sql']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when passed an empty array', () => {
    const tree = renderer
      .create(<SearchSuggestion suggestedTerms={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });  
});
