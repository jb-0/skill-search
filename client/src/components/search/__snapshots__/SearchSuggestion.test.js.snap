// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTerms component renders correctly when passed an array containing suggested terms 1`] = `
<div>
  <div
    className="suggested-term-container"
    data-testid="suggested-term-container"
    id="node"
  >
    <p
      id="node"
    >
      node
    </p>
  </div>
  <div
    className="suggested-term-container"
    data-testid="suggested-term-container"
    id="sql"
  >
    <p
      id="sql"
    >
      sql
    </p>
  </div>
</div>
`;

exports[`SearchTerms component renders correctly when passed an empty array 1`] = `<div />`;
